TrafficSimulator/Assets/Cam/DefaultCamera.cs

```diff
@@ -0,0 +1,151 @@
+            }
+            else
+            {
+                if(_hasToggledGameObject) SetToggledGameObjectToNull();
+            }
+        }
+
+        public override void HandleDoubleClickInput(InputAction.CallbackContext ctx)
+        {
+            var ray = Camera.main.ScreenPointToRay(Mouse.current.position.ReadValue());
+            if (Physics.Raycast(ray, out var hitInfo) && hitInfo.transform.gameObject.Equals(_toggledGameObject))
+            {
+                CameraManager.CameraTarget = FollowTransform;
+                CameraManager.ToggleThirdPersonCamera();
+            }
+        }
+
+        public override void Move(Vector3 direction)
+        {
+            // Ignore the input argument if the mouse is near the screen border
+            if (_isNearScreenBorder) direction = _mousePointDirection.normalized;
+
+            if (direction.sqrMagnitude != 0 && _hasToggledGameObject) SetToggledGameObjectToNull();
+            Vector3 translatedDirection = TranslateDirectionToForward(direction.y, direction.x);
+            FollowTransform.position += translatedDirection * (_movementSpeed * Time.deltaTime);
+        }
+
+        public override void RotateHorizontal(float horizontalRotation)
+        {
+            FollowTransform.eulerAngles += new Vector3(0, horizontalRotation * _rotationSpeed * Time.deltaTime, 0);
+        }
+
+        public override void Zoom(float zoomValue)
+        {
+            _targetZoom -= zoomValue;
+            _targetZoom = Mathf.Clamp(_targetZoom, _maxZoom, _minZoom);
+            VirtualCamera.m_Lens.FieldOfView = Mathf.Lerp(VirtualCamera.m_Lens.FieldOfView,
+                _targetZoom, Time.deltaTime * _zoomSpeed);
+        }
+        
+        private void SetToggledGameObjectToNull()
+        {
+            _toggledGameObject.GetComponent<Outline>().enabled = false;
+            _toggledGameObject = null;
+            _hasToggledGameObject = false;
+        }
+
+        private void SetToggledGameObject(GameObject toggledGameObject)
+        {
+            Outline outline = toggledGameObject.GetComponent<Outline>();
+            outline.enabled = true;
+            _toggledGameObject = toggledGameObject;
+            _hasToggledGameObject = true;
+            StartCoroutine(PanToTarget(toggledGameObject.transform.position));
+        }
+        
+        private Vector3 TranslateDirectionToForward(float forwardScalar, float sidewaysScalar)
+        {
+            return FollowTransform.forward * forwardScalar + transform.right * sidewaysScalar;
+        }
+
+        IEnumerator PanToTarget(Vector3 target)
+        {
+            _isMovingTowardsToggledObject = true;
+            float startTime = Time.time;
+            float journeyLength = Vector3.Distance(FollowTransform.position, _toggledGameObject.transform.position);
+
+            while (Vector3.Distance(FollowTransform.transform.position, target) > 0.01f && _hasToggledGameObject)
```

Comment:
Is this supposed to access the transform of a transform? Isn't that the same as just using `FollowTransform.position` directly? I thought `FollowTransform.transform` would be the same as `FollowTransform`